fs/ftpfs.py

```diff
@@ -328,6 +331,22 @@ def __str__(self):
         )
         return _fmt.format(host=self.host, port=self.port)
 
+    @property
+    def user(self):
+        return (
+            self._user
+            if self.proxy is None else
+            '{}@{}'.format(self._user, self._host)
+        )
+
+    @property
+    def host(self):
+        return (
+            self.proxy
+            if self.proxy is not None else
+            self._host
+        )
+
```

Comment:
I prefer using shorter code like 
```
return self.proxy or self._host
```
But I also understand that you wanted to keep user() and host() alike. If that was the goal you wanted to achieve I would at least change one of
```
if self.proxy is None else
```
and
```
if self.proxy is not None else
```
in order to be consistent. Maybe change host() to this?
```
def host(self):
        return (
            self._host
            if self.proxy is None else
            self.proxy
        )
```

Reply:
> return self.proxy or self._host

In this case I want to be explicit regarding about using `None` rather than anything that evaluates to False.

> in order to be consistent. Maybe change host() to this?

:+1: